HttpClientFactory Keyed DI docs (#44533)
dotnet/docs: docs/core/extensions/httpclient-factory.md

For more information, see the [full example](https://github.com/dotnet/docs/tree/main/docs/core/extensions/snippets/http/scopeworkaround).

## Avoid depending on "factory-default" Primary Handler

> In this section, we use the term _"factory-default" Primary Handler_, by which we mean the Primary Handler that the default `IHttpClientFactory` implementation (or more precisely, the default `HttpMessageHandlerBuilder` implementation) assigns if _not configured in any way_ whatsoever.

Reviewer: ```suggestion
In this section, the term _"factory-default" Primary Handler_ refers to the Primary Handler that the default `IHttpClientFactory` implementation (or more precisely, the default `HttpMessageHandlerBuilder` implementation) assigns if _not configured in any way_ whatsoever.
```

Reviewer: Style rules want us to avoid we/our in docs - https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/w/we?toc=%2Fwriting-style-guide-msft-internal%2Ftoc.json&bc=%2Fwriting-style-guide-msft-internal%2Fbreadcrumb%2Ftoc.json

> [!NOTE]
> The "factory-default" Primary Handler is an _implementation detail_ and subject to change.
> ❌ AVOID depending on a specific implementation being used as a "factory-default" (for example, `HttpClientHandler`).

> There are cases in which you need to know the specific type of a Primary Handler, especially if working on a class library. While preserving the end-user's configuration, you might want to update, for example, `HttpClientHandler`-specific properties like `ClientCertificates`, `UseCookies`, `UseProxy`, etc. It might be tempting to cast the Primary handler to `HttpClientHandler`, which _happened to_ work while `HttpClientHandler` was used as the "factory-default" Primary Handler. But as any code depending on implementation details, such a workaround is _fragile_ and bound to break.

Reviewer: ```suggestion
There are cases in which you need to know the specific type of a Primary Handler, especially if working on a class library. While preserving the end user's configuration, you might want to update, for example, `HttpClientHandler`-specific properties like `ClientCertificates`, `UseCookies`, and `UseProxy`. It might be tempting to cast the Primary handler to `HttpClientHandler`, which _happened to_ work while `HttpClientHandler` was used as the "factory-default" Primary Handler. But as any code depending on implementation details, such a workaround is _fragile_ and bound to break.
```
